### ISSUE TITLE
Running behind https Apache proxy: cross domain issues?

### ISSUE THREAD

**Issue by rduivenvoorde** [flobz/psa_car_controller#304]
Thanks very much for this!! Just started to play with it. Before I had a Leaf of which I requested the battery level every 30 minutes...

I'm on Debian (server and desktop) and driving an Opel Corsa E now

I succesfully run the app locally, seeing/controlling over http://localhost:5000

Next try was to run it on my homeserver behind an (Apache) reverse proxy, THAT partially works:

- I run it on 5001 (because another small script is using 5000):  `python3 server.py --web-conf -p 5001 -d -r`
- configure Apache to pass request from `/foo` to the app
```  
   ProxyPass /foo http://localhost:5001
   ProxyRequests Off
   ProxyPassReverse / https://myserver.net/foo
```
- had troubles starting dash, failed to use config.json and got things running by adding `requests_pathname_prefix` in `web/app.py` : 
```
    dash_app = DashCustom(external_stylesheets=[dbc.themes.BOOTSTRAP], external_scripts=locale_url, title=title,
                          server=app, requests_pathname_prefix="/foo/",
                          suppress_callback_exceptions=True, serve_locally=False)
```
- still had some issues retrieving assets so created a symlink to `web\assets\ in my root
- but: all is looking fine, except one error:

![Screenshot-20211226114916-632x253](https://user-images.githubusercontent.com/731673/147405835-7e7b3205-2470-461e-82bc-ca750c039c23.png)
 
- looking into the web browser console, I see that (Dash?) is trying to retrieve files from the http-version of my httpS://myserver.net app:
```Blocked loading mixed active content “http://myserver.net/control?header=false”``` (should look on `https://myserver.net/foo/control`  I think?

So my question: anybody else succesfully running the controller behind a proxy?

Should I use config.json or other startup params to fix the requests_pathname_prefix ?

Other hints?

Another thing I see in running the server when requesting the get_vehicleinfo (which runs fine, so not sure if that is related):

```
2021-12-26 11:56:38,379 :: DEBUG :: battery table is probably empty :
Traceback (most recent call last):
  File "/home/richard/dev/psa_car_controller/libs/charging.py", line 60, in record_charging
    start_at, stop_at, start_level = conn.execute(
TypeError: cannot unpack non-iterable NoneType object
2021-12-26 11:56:38,380 :: DEBUG :: charging_status:Disconnected 
```

**Comment by flobz** [comment]
You need to set /foo as prefix at program startup.
Do server.py - h to see how to do it.

**Comment by rduivenvoorde** [comment]
Ah, duh.. sorry for missing that :-(

Ok, I think I found the crux for the cross domain issues I saw in the javascript console (for those running behind an Apache reverse proxy).


You really have to set this:
```
RequestHeader set X-Forwarded-Proto https
```
as THAT was the reason I got a lot cross-domain issues: the Apache (running httpS) was forwarding to the Dash/Flask server which received the requests as http (without S) and got confused. 

The above header fixed all my issues (in combination with the -d option!)

Thanks!
